src/core/blueprint/operators/graphql.rs

```diff
@@ -65,17 +75,27 @@ pub fn compile_graphql(
     Valid::succeed(graphql.url.as_str())
         .zip(helpers::headers::to_mustache_headers(&graphql.headers))
         .and_then(|(base_url, headers)| {
-            Valid::from(
-                RequestTemplate::new(
-                    base_url.to_owned(),
-                    operation_type,
-                    &graphql.name,
-                    args,
-                    headers,
-                    create_related_fields(config, type_name, &mut HashSet::new()),
-                )
-                .map_err(|e| ValidationError::new(e.to_string())),
+            Valid::from_option(
+                create_related_fields(
+                    config,
+                    VecDeque::from(vec![type_name.to_string(); graphql.depth.unwrap_or(5)]),
+                    &mut HashSet::new(),
+                ),
+                "Logical error occurred while creating Related Fields".to_string(),
             )
+            .and_then(|related_fields| {
+                Valid::from(
+                    RequestTemplate::new(
+                        base_url.to_owned(),
+                        operation_type,
+                        &graphql.name,
+                        args,
+                        headers,
+                        related_fields,
+                    )
+                    .map_err(|e| ValidationError::new(e.to_string())),
+                )
+            })
```

Comment:
This implementation needs to be changed. The related field information can't be inferred while creating a blueprint. It should be implemented as a part of the JIT transformation step. The JIT Transformations should identify exactly what the RequestTemplate should look like for each query and then while evaluating we should simply use that operation and execute the GraphQL request.